### PR TITLE
Working on comments

### PR DESCRIPTION
**Pull request by MalikOJBS**
Comments can be posted and show under posts

### CODE REVIEW

File: controllers/posts.js [hunk 1]
```diff
@@ -61,6 +61,28 @@ const PostsController = {
...
+  Comment: function(req, res) {
+    Post.find({_id: req.params._id}, function(err, posts) {
+      if (err) {
+        throw err;
+      }
+      res.render('posts/comment', {
+        posts: posts});
+    });
+  },
+
```
Comment (kaykakaraka) [inline controllers/posts.js:73]: do we ever use posts/comments?

File: views/posts/index.hbs [hunk 2]
```diff
@@ -19,9 +19,21 @@
...
+  <p> Comments: </p>
+  <div class='array-output-wrapper'>
+    <ul> {{#each this.comments}}
+        <li> {{this}} </li>
+      {{/each}}
+  </div>
   
+  {{/each}}
+
 </ul>
```
Comment (kaykakaraka) [inline views/posts/index.hbs:38]: looks good

File: controllers/posts.js [hunk 3]
```diff
@@ -61,6 +61,28 @@ const PostsController = {
...
+    Post.findOneAndUpdate({
+      _id: req.params._id},
+    {$push: {comments: req.body.comments}},
+    function(err, posts) {
+      if (err) {
+        throw err;
+      }
+    res.status(201).redirect('/posts');
+    });
+  },
```
Comment (kaykakaraka) [inline controllers/posts.js:84]: a smooth way of doing this

**Review by kaykakaraka — CHANGES_REQUESTED** [review]
I'm wondering if we can remove the Comment section in the controller, and the route to it? I'm not sure we ever use it (I could be misunderstanding how it works).

Looks good though! Very efficient code

File: controllers/posts.js [hunk 4]
```diff
@@ -61,6 +61,32 @@ const PostsController = {
     });
   },
 
+  Comment: function(req, res) {
+    Post.find({_id: req.params._id}, function(err, post) {
+      console.log(err)
+      console.log("///")
+      console.log(post)
```
Comment (michaelcychan) [inline controllers/posts.js:68]: these console.log can be removed if the implementation is complete

File: cypress/integration/comment_box_scpec.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// describe("Timeline", function() {
```
Comment (michaelcychan) [inline cypress/integration/comment_box_scpec.js:1]: This whole test is commented, do we need to un-comment this test?

File: spec/models/post.spec.js [hunk 6]
```diff
@@ -73,5 +73,26 @@ describe("Post model", () => {
...
+          {'message': 'some message'},
+          {$set: {'comments': 'amazing'}},
+          function(err) {
+            expect(err).toBeNull();
+
+        Post.find(function(err, posts) {
+          expect(err).toBeNull();
+
+          expect(posts[0]).toMatchObject({
+            message: 'some message'});
```
Comment (michaelcychan) [inline spec/models/post.spec.js:92]: I suggest we need to add one more expect statement, something like:
expect(posts[0].comments[0]).toEqual('amazing');

Currently the unit test does not test for the comment itself

File: views/posts/index.hbs [hunk 7]
```diff
@@ -19,9 +19,27 @@
...
+
+<p> Comments: </p>
+<div class='array-output-wrapper'>
+  <ul id= "comments-list">
+    {{#each this.comments}}
+      <li id = "commenting"> {{this}} </li>
+    {{/each}}
+    </div>
+  </div>
+
```
Comment (michaelcychan) [inline views/posts/index.hbs:40]: simple and easy to read. nice

File: views/posts/comment.hbs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+{{#each posts}}
+<form id="comment-post-form" action="/posts/{{this._id}}/comment" method="post">
+  <label for="comments">Write your comment: </label>
+  <input id="comments" type="text" name="comments" value="">
+  <input type="submit" value="Submit">
+</form>
+
+{{/each}}
```
Comment (michaelcychan) [inline views/posts/comment.hbs:8]: nice

**Review by michaelcychan — CHANGES_REQUESTED** [review]
integration test and unit test need some changes

File: cypress/integration/comment_box_spec.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+describe("Timeline", function() {
+  it("can can view comments on post", function() {
```
Comment (michaelcychan) [inline cypress/integration/comment_box_spec.js:2]: clear comments to explain the test parts!

**Review by michaelcychan — APPROVED** [review]
Great job!

File: controllers/posts.js (hunk 1 above)
Comment (Hafizalimi7) [reply controllers/posts.js:77]: it is now it use, thanks

**Review by kaykakaraka — APPROVED** [review]

File: controllers/posts.js (hunk 4 above)
Comment (Hafizalimi7) [reply controllers/posts.js:68]: changes have been made

File: cypress/integration/comment_box_scpec.js (hunk 5 above)
Comment (Hafizalimi7) [reply cypress/integration/comment_box_scpec.js]: it has now been un-commented

File: cypress/integration/comment_box_scpec.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+//   it("can can view comments on post", function() {
+//     // sign up
+//     cy.visit("/users/new");
```
[the remark this answers is not in the document]
Comment (Hafizalimi7) [reply cypress/integration/comment_box_scpec.js]: sorted

File: views/posts/index.hbs (hunk 2 above)
Comment (Hafizalimi7) [reply views/posts/index.hbs]: thanks

File: spec/models/post.spec.js (hunk 6 above)
Comment (Hafizalimi7) [reply spec/models/post.spec.js]: sorted

**Pull request merged**